### ISSUE TITLE
Add support for saving/favoriting an image 

### ISSUE THREAD

**Issue by noahmalmed** [zooniverse/mobile#183]
We have had a couple of requests recently to add the ability to save or favorite images on the app.

I think this could just be as simple as adding a share button or a favorite heart or something.

@beckyrother thoughts?

**Comment by beckyrother** [comment]
@noahmalmed should this include adding to a collection as well?

**Comment by noahmalmed** [comment]
We could, although adding a collection option would be significantly more work. Plus a user could add it to a collection from the favorite screen on the website. I think favorite and download are more important

**Comment by beckyrother** [comment]
I've always considered Favorites and Collections connected, and we don't even offer image downloads for desktop. Maybe we should pose this question to the rest of the team

**Comment by astopy** [comment]
There will be project owners who won't like us giving users an option to save/download images (even if there isn't anything technically stopping users from doing that now).

**Comment by srallen** [comment]
I agree with @astopy. I had to hide the download button option that Chrome exposed on the HTML 5 video player for Chimp&See using some hacky CSS a few months ago. Although savvy users can figure out how to download the media files anyway, showing a download button could imply a perception of approval, consent, or encouragement to do so by the project. With Chimp&See, they do not want users downloading the videos to help project the site location from poachers. 

As for favorites, they technically are a kind of collection resource with just the `favorite: true` property set and linked to a project. Check out how we do it on PFE: https://github.com/zooniverse/Panoptes-Front-End/blob/c14f06f7aa93949e2a9afa307db96b9267d4a74b/app/collections/favorites-button.jsx

You have to check if a favorites collection already exists for the project and if it doesn't, create one if the button is hit.

The full collection feature would involve also supporting the POST to create a new collection linked to the project as well as a search box for existing. I think in this case just for an initial support of the feature, just adding the favorite button would be ok and then adding the additional custom collection creation/add feature could be next if users want it in the mobile app.

**Comment by noahmalmed** [comment]
Thanks @srallen and @astopy for the feedback on this. 

@beckyrother how does adding a favorite option as a first pass at this sound?

**Comment by beckyrother** [comment]
Sounds good to me!

**Comment by noahmalmed** [comment]
Designs are available for this [on invision](https://projects.invisionapp.com/d/main#/console/13393648/333632079/preview)

**Comment by noahmalmed** [comment]
Take a look at `SwipeCard` and `SubjectOptionsBar` to get started on these changes
